src/chordChart.ts

```diff
@@ -71,1075 +81,1300 @@ import translateAndRotate = manipulation.translateAndRotate;
-
-                    tooltipInfo.push(ChordChart.createTooltipInfo(
-                        this.data.groups,
-                        this.data.dataMatrix,
-                        tooltipEvent.data.target));
-                }
-
-                return tooltipInfo;
-            });
+    for (let labelDataPoint of labelDataPoints) {
+      let data: ChordArcLabelData = labelDataPoint.data;
+
+      let colorInstance: VisualObjectInstance = {
+        objectName: "dataPoint",
+        displayName: data.label,
+        selector: ColorHelper.normalizeSelector(
+          (<ISelectionId>labelDataPoint.identity).getSelector()
+        ),
+        properties: {
+          fill: { solid: { color: data.barFillColor } },
+        },
+      };
+
+      if (data.isCategory || data.isGrouped) {
+        instances.push(colorInstance);
+      }
     }
 
-    private static createTooltipInfo(labelDataPoints: ChordArcDescriptor[], dataMatrix: number[][], source: any) {
-        return {
-            displayName: labelDataPoints[source.index].data.label
-                + "->" + labelDataPoints[source.subindex].data.label,
-            value: dataMatrix[source.index][source.subindex].toString()
-        };
+    return instances;
+  }
+
+  private static enumerateLabels(
+    settings: Settings,
+    localManager: ILocalizationManager
+  ): VisualObjectInstance[] {
+    const labelSettings: LabelsSettings = settings.labels;
+
+    return [
+      {
+        objectName: "labels",
+        displayName: localManager.getDisplayName("Visual_Labels"),
+        selector: null,
+        properties: {
+          show: labelSettings.show,
+          color: labelSettings.color,
+          fontSize: labelSettings.fontSize,
+        },
+      },
+    ];
+  }
+
+  // Calculate radius
+  private calculateRadius(): number {
+    if (this.settings.labels.show) {
+      // if we have category or data labels, use a sigmoid to blend the desired denominator from 2 to 3.
+      // if we are taller than we are wide, we need to use a larger denominator to leave horizontal room for the labels.
+      let hw: number =
+        this.layout.viewportIn.height / this.layout.viewportIn.width;
+      let denom: number = 2 + 1 / (1 + Math.exp(-5 * (hw - 1)));
+      return (
+        Math.min(this.layout.viewportIn.height, this.layout.viewportIn.width) /
+        denom
+      );
     }
 
-    private clear(): void {
-        this.clearNodes([
-            ChordChart.chordClass,
-            ChordChart.sliceClass,
-            ChordChart.sliceTicksClass,
-            ChordChart.labelsClass,
-            ChordChart.lineClass
-        ]);
-    }
+    // no labels
+    return (
+      Math.min(this.layout.viewportIn.height, this.layout.viewportIn.width) / 2
+    );
+  }
+
+  private drawCategoryLabels(): void {
+    // Multiplier to place the end point of the reference line at 0.05 * radius away from the outer edge of the chord/pie.
+    let arc: Arc<any, d3.DefaultArcObject> = d3
```

Comment:
this is d3 package. Replace it by a smaller package

Reply:
done